manager/src/main/java/org/openremote/manager/dashboard/DashboardStorageService.java

```diff
@@ -68,36 +67,40 @@ public void stop(Container container) throws Exception {
 
     /* --------------------------  */
 
-    // Getting ALL dashboards from a realm
-    protected <T extends Dashboard> Dashboard[] findAllOfRealm(String realm, String userId) {
-        return this.findAllOfRealm(realm, userId, false);
-    }
-
-    @SuppressWarnings("java:S2326")
-    protected <T extends Dashboard> Dashboard[] findAllOfRealm(String realm, String userId, Boolean editable) {
+    // Querying dashboards from the database
+    // userId is required for checking dashboard ownership. If userId is NULL, we assume the user is not logged in.
+    // canEdit can be used to only return editable dashboards when 'true'.
+    protected <T extends Dashboard> Dashboard[] query(String dashboardId, String realm, String userId, Boolean canEdit) {
         return persistenceService.doReturningTransaction(em -> {
             try {
                 CriteriaBuilder cb = em.getCriteriaBuilder();
                 CriteriaQuery<Dashboard> cq = cb.createQuery(Dashboard.class);
                 Root<Dashboard> root = cq.from(Dashboard.class);
+
                 List<Predicate> predicates = new ArrayList<>();
+                if(dashboardId != null) {
+                    predicates.add(cb.like(root.get("id"), dashboardId));
+                }
                 if(realm != null) {
                     predicates.add(cb.like(root.get("realm"), realm));
                 }
```

Comment:
Not sure it should be possible to query dashbaords across realms so this parameter should expect not null or throw

Reply:
Don't see a change here....